docs/1_CODING_CONVENTIONS.md

```diff
@@ -0,0 +1,268 @@
+# Coding Conventions
+
+Here you can read about our preferred ways of writing code.
+
+## Modern JavaScript
+
+Use [Babel](https://babeljs.io/) to transpile your code from modern JavaScript
+to traditional JavaScript that will run in any browser.
+
+### Classes
+
+Use [ES2015 classes](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes)
+instead of prototype-based inheritance. Having a single declarative form makes
+class patterns easier to use, and encourages interoperability.
+
+```javascript
+class Todo {
+  constructor(text) {
+    this.text = text
+  }
+}
+```
+
+### Arrow functions
+
+Use [arrow functions](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+instead of traditional functions wherever possible.
+
+Arrow functions are especially useful in class methods, because arrow functions
+share the same lexical **this** with the surrounding code, so no binding is
+required.
+
+**Do**
+
+```javascript
+class Todos {
+  addTodo = (todo) => {
+    // ...
+  }
+}
+```
+
+**Do NOT do**
+
+```javascript
+class Todos {
+  constructor() {
+    this.addTodo = this.addTodo.bind(this);
+  }
+
+  addTodo(todo) {
+    // ...
+  }
+}
+```
+
+What comes to functions, arrow functions are less verbose and more convenient
+than traditional functions.
+
+**Do**
+
+```javascript
+const addTodo = () => {
+  // ...
+}
+```
+
+**Do NOT do**
+
+```javascript
+function addTodo() {
+  // ...
+}
+```
+
+### Template literals
+
+Use [template literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals)
+instead of string concatenation. Template literals allow for embedded
+expressions and you can use them even with multi-line strings.
+
+**Do**
+
+```javascript
+const hello = `Hello ${name}!
+This syntax is convenient.`
+```
+
+**Do NOT do**
+
+```javascript
+const hello = 'Hello ' + name + '!' +
+'This syntax is not so convenient.'
+```
+
+### Destructing assignment
+
+Use [destructing assignment](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment)
+wherever possible. Destructing allows for a shorter syntax that is easier to read.
```

Comment:
I'd say use it within reason. Some components take too many props for regular destructuring to be readable.